tc_solaris.go

```diff
@@ -0,0 +1,75 @@
+package console
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func tcget(fd uintptr, p *unix.Termios) error {
+	var err error
+	p, err = unix.IoctlGetTermios(int(fd), unix.TCGETS)
```

Comment:
i think this doesnt work, you should swap the value instead, check `tc_linux`.

Also, is solaris any different from linux regarding to these functions ? maybe we can just have buildtags instead of duplicating the code ?